CMakeLists.txt

```diff
@@ -4,6 +4,48 @@ project(ozo VERSION 0.0.1)
 
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules")
 
+include(CheckCXXCompilerFlag)
+
+function (target_supported_compile_options target)
```

Comment:
This is a good idea, but I think it would be nice to point certain flags for the certain compilers to be sure about flags set-up.

Reply:
Well, I meant it would be better to have an error if a compiler does not support an option. I would like to say what unsupported option should not be ignored silently because in that case, you can not have a guarantee that a given option is supported. E.g., in the case of a typo, there should be an error. So the silent strategy is not good in this case.

Reply:
While I don't necessarily agree (I personally think it's nicer to do "feature detection" than "compiler detection"), I have changed it now.

Reply:
Well, detection is a good idea, but if a feature is not supported, but needed - build should fail. In other cases - it is equivalent to the absence of a feature. Since different compilers require different features it is needed to bind feature set to the certain compiler(s). In other cases, requirements are just meaningless because no checks could be made.
To rely just on compiler errors is not a good strategy in terms of error source detection. Since sometimes it is too hard to understand relation about missed feature and compilation error.